Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+            if (conn.playerController.gameObject.GetComponent<NetworkLobbyPlayer>().ReadyToBegin)
+                countPlayers += 1;
+
+            return countPlayers;
+        }
+
+        public void CheckReadyToBegin()
+        {
+            if (SceneManager.GetActiveScene().name != LobbyScene) return;
+
+            int readyCount = 0;
+
+            foreach (KeyValuePair<int, NetworkConnection> conn in NetworkServer.connections)
+            {
+                if (conn.Value == null)
+                    continue;
+
+                readyCount += CheckConnectionIsReadyToBegin(conn.Value);
+            }
+
+            if (minPlayers > 0 && readyCount < minPlayers)
+            {
+                allPlayersReady = false;
+                return;
+            }
+
+            pendingPlayers.Clear();
+            allPlayersReady = true;
+            OnLobbyServerPlayersReady();
+        }
+
+        public void ServerReturnToLobby()
+        {
+            if (!NetworkServer.active)
+            {
+                Debug.Log("ServerReturnToLobby called on client");
+                return;
+            }
+            ServerChangeScene(LobbyScene);
+        }
+
+        void CallOnClientEnterLobby()
+        {
+            OnLobbyClientEnter();
+            foreach (NetworkLobbyPlayer player in lobbySlots)
+            {
+                if (player == null)
+                    continue;
+
+                player.OnClientEnterLobby();
+            }
+        }
+
+        void CallOnClientExitLobby()
+        {
+            OnLobbyClientExit();
+            foreach (NetworkLobbyPlayer player in lobbySlots)
+            {
+                if (player == null)
+                    continue;
+
+                player.OnClientExitLobby();
+            }
+        }
+
+        #region server handlers
+
+        public override void OnServerConnect(NetworkConnection conn)
+        {
+            if (numPlayers >= maxConnections)
+            {
+                conn.Disconnect();
+                return;
+            }
+
+            // cannot join game in progress
+            if (SceneManager.GetActiveScene().name != LobbyScene)
+            {
+                conn.Disconnect();
+                return;
+            }
+
+            base.OnServerConnect(conn);
+            OnLobbyServerConnect(conn);
+        }
+
+        public override void OnServerDisconnect(NetworkConnection conn)
+        {
+            if (conn.playerController != null)
+            {
+                NetworkLobbyPlayer player = conn.playerController.GetComponent<NetworkLobbyPlayer>();
+
+                if (player != null)
+                    lobbySlots.Remove(player);
+            }
+
+            allPlayersReady = false;
+
+            foreach (NetworkLobbyPlayer p in lobbySlots)
+            {
+                if (p != null)
+                    p.GetComponent<NetworkLobbyPlayer>().ReadyToBegin = false;
+            }
+
+            base.OnServerDisconnect(conn);
```

Comment:
please add comment on why base.onserverdisconnect is called here and not before/after

Reply:
Moved to bottom...I think it needed to be there because of something else at one point but not anymore.